crates/env/src/call/call_builder.rs

```diff
@@ -597,9 +657,23 @@ where
     E: Environment,
 {
     /// Invokes the cross-chain function call.
+    ///
+    /// # Panics
+    ///
+    /// This method panics if it encounters an [`ink_primitives::LangError`]. If you want to handle
+    /// those use the [`try_fire`][`CallBuilder::try_fire`] method instead.
     pub fn fire(self) -> Result<(), Error> {
         self.params().invoke()
     }
+
+    /// Invokes the cross-chain function call.
+    ///
+    /// # Note
+    ///
+    /// On failure this returns an [`ink_primitives::LangError`] which can be handled by the caller.
```

Comment:
It would be more accurate to clarify that the *inner* Result will be a `LangError`, and the outer Result is an environmental error.

Reply:
Yep, I'll fix it in the follow-up that removes the `EnvError` from the default execution path